src/console_op/consoleOpMain.go

```diff
@@ -281,17 +281,23 @@ func main() {
 		log.Print("Running in DEBUG-ONLY mode.")
 	}
 
+	// construct dependency injection
+	k8Manager, err := NewK8Manager()
+	if err != nil {
+		log.Printf("ERROR: k8Manager failed to initialize")
+	}
+	nodeManager := NewNodeManager(k8Manager)
+	dataManager := NewDataManager(k8Manager)
+	healthManager := NewHealthManager(dataManager)
+
 	// Set up the zombie killer
 	go watchForZombies()
 
-	// initialize the connection with k8s client
-	initK8s()
-
 	// loop over new hardware
-	go watchHardware()
+	go watchHardware(dataManager, nodeManager)
```

Comment:
You seem to use the terms manager and service interchangeably, such as here where you pass in dataManager but watchHardware on line 201 takes a dataService and calls the object "ds" rather than "dm".  I understand the difference between the struct and the interface, I'm just trying to understand if there's a difference between the terms DataManager and DataService in your mind.

Reply:
The method signatures expect the interface type so I am able to pass in the concrete struct (manager) just naming convention I have used before with dependency injection. Technically they are the same since the manager is just implementing the interface.

This provides the ability to 1 keep the contract of the interface strongly typed, if we add a new API to the interface that should be implemented in the manager the dependent code will complain. Another point is now we can instead pass another struct that implements the interface type so in my unit test I do just that.